src/librbd/Operations.cc

```diff
@@ -342,14 +342,6 @@ int Operations<I>::flatten(ProgressContext &prog_ctx) {
     return -EROFS;
   }
 
-  {
```

Comment:
This is here to prevent recording a failure in the journal

Reply:
Hi, Jason
I don't really understand how this thunk is different from the same part of execute_flatten , could you give me some hints?
Thanks in advance.

Reply:
Hi, Jason
I don't really understand how this thunk is different from the same part of execute_flatten, could you give me some hints?
Thanks in advance.

Reply:
It executes before an op start event is recorded in the journal (when journaling is enabled) to allow it to quickly fail. If you look throughout this file, you will see each operation doing a duplicate check before it starts the op.

Reply:
Yes, exactly, thanks for the detailed clarification and sorry for the noise, looks like this PR is redundant, I will close it later.